share/spice/gifs/gifs.js

```diff
@@ -1,31 +1,43 @@
-function ddg_spice_gifs(res) {
-    if(!res || !res.data || !res.data.length){ return; }
+(function(env){
+    "use strict";
 
-    var searchTerm = DDG.get_query().replace(/gifs?/i,'').trim();
+    env.ddg_spice_gifs = function (api_result) {
 
-    Spice.add({
-        id: 'gifs',
-        name: 'Gifs',
+        if (!api_result || !api_result.data || !api_result.data.length){
+            return Spice.failed("gifs");
+        }
 
-        data: res.data,
-        normalize: function(item) {
-            return {
-                thumbnail: item.images.fixed_height_still.url,
-                image: item.images.fixed_height.url,
-                url: item.url,
-                height: item.images.fixed_height.height,
-                width: item.images.fixed_height.width
-            };
-        },
-        meta: {
-            sourceName: 'Giphy',
-            sourceUrl: 'http://giphy.com/search/' + encodeURIComponent(searchTerm),
-            sourceIcon: true,
-            count: res.pagination.count,
-            total: res.pagination.total_count,
-            itemType: 'Gifs'
-        },
+        var searchTerm = DDG.get_query().replace(/gifs?|giphy/ig,'').trim();
```

Comment:
Maybe shouldn't be `//g`? There's the case where you could search for "giphy gifs" and it would strip out both of them.